File: p002/p002.cpp
#include <iostream>
using namespace std;
        
int main() {
    const int limit = 4 * 1000 * 1000;
    int s = 0;
    int n = 0;
    int n1 = 0;
    int n2 = 1;
    while(n <= limit) {
        n = n1 + n2;
        n1 = n2;
        n2 = n;
        if(n % 2 == 0) {
            s += n;
        }
    }
    cout << s << endl;
    return 0;
}

File: p002/p002.rb
n = 0
n1 = 0
n2 = 1
s = 0
limit = 4 * 1000 * 1000
loop do
    n = (n1 + n2)
    n1 = n2
    n2 = n
    s += n if (n % 2) == 0
    break if n > limit
end

puts s
